pkg/core/store.go

```diff
@@ -106,11 +106,27 @@ func (s *StoreInfo) Clone(opts ...StoreCreateOption) *StoreInfo {
 	store := *s
 	store.meta = typeutil.DeepClone(s.meta, StoreFactory)
 	for _, opt := range opts {
-		opt(&store)
+		if opt != nil {
```

Comment:
When will it become nil?

Reply:
https://github.com/tikv/pd/blob/ef2a01f7bcacd4074541cf4131c8a8f667a0604e/server/cluster/cluster.go#L872 in this case, the option maybe nil

Reply:
https://github.com/tikv/pd/blob/ef2a01f7bcacd4074541cf4131c8a8f667a0604e/server/cluster/cluster.go#L872 in this case, the option maybe nil